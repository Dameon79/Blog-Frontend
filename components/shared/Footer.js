import { Component } from 'react'
import Link from 'next/link'

class Footer extends Component {
  render () {
    return (
      <footer className='c-footer'>
        <nav className='c-footer__nav'>
          <Link href='/'>
            <a className='c-burgernav__menu'>Home</a>
          </Link>
          <Link href='/about' as='about-me'>
            <a className='c-burgernav__menu'>About Me</a>
          </Link>
          <Link href='/projects' as='my-projects'>
            <a className='c-burgernav__menu'>My Projects</a>
          </Link>
          <Link href='/articles' as='all-blog-posts'>
            <a className='c-burgernav__menu'>Blog Articles</a>
          </Link>
          <Link href='/contact' as='contact-me'>
            <a className='c-burgernav__menu u-white-border'>Contact</a>
          </Link>
        </nav>

        <nav className='c-footer__socialmedia'>
          <a href='https://github.com/Dameon79'>
            <img src='/static/images/Github2.png' className='c-socialmedialogos c-socialmedialogos__git' />
          </a>
          <a href='https://procoders.co.uk/'>
            <img src='/static/images/procoders2.jpg' className='c-socialmedialogos' />
          </a>
          <a href='https://www.linkedin.com/in/dameon-sumpton-407503ab/'>
            <img src='/static/images/linkedin1.png' className='c-socialmedialogos' />
          </a>
          <a href='https://twitter.com/DameonSumpton'>
            <img src='/static/images/Twitter1.png' className='c-socialmedialogos' />
          </a>
        </nav>
        <h6 className='c-footer__copyright'>&copy; Dameon Sumpton {new Date().getFullYear()} </h6>
      </footer>
    )
  }
}

export default Footer
