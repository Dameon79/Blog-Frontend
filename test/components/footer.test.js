import Footer from '../../components/shared/Footer'

describe('<Footer />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Footer />
    )

    expect(component).toMatchSnapshot()
  })
})