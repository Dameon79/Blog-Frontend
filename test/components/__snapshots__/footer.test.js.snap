// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="c-footer__nav"
        >
          <Link
            href="/index"
          >
            <a
              className="c-burgernav__menu"
            >
              Home
            </a>
          </Link>
          <Link
            href="/about"
          >
            <a
              className="c-burgernav__menu"
            >
              About Me
            </a>
          </Link>
          <Link
            href="/projects"
          >
            <a
              className="c-burgernav__menu"
            >
              My Projects
            </a>
          </Link>
          <Link
            href="/articles"
          >
            <a
              className="c-burgernav__menu"
            >
              Blog Articles
            </a>
          </Link>
          <Link
            href="/contact"
          >
            <a
              className="c-burgernav__menu u-white-border"
            >
              Contact
            </a>
          </Link>
        </nav>,
        <nav
          className="c-footer__socialmedia"
        >
          <a
            href="https://github.com/Dameon79"
          >
            <img
              className="c-socialmedialogos c-socialmedialogos__git"
              src="/static/images/Github2.png"
            />
          </a>
          <a
            href="https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE"
          >
            <img
              className="c-socialmedialogos"
              src="/static/images/procoders2.jpg"
            />
          </a>
          <a
            href="https://www.linkedin.com/in/dameon-sumpton-407503ab/"
          >
            <img
              className="c-socialmedialogos"
              src="/static/images/linkedin1.png"
            />
          </a>
          <a
            href="https://twitter.com/dameon_sumpton"
          >
            <img
              className="c-socialmedialogos"
              src="/static/images/Twitter1.png"
            />
          </a>
        </nav>,
        <h6
          className="c-footer__copyright"
        >
          Â© Dameon Sumpton 2018
        </h6>,
      ],
      "className": "c-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href="/index"
            >
              <a
                className="c-burgernav__menu"
              >
                Home
              </a>
            </Link>,
            <Link
              href="/about"
            >
              <a
                className="c-burgernav__menu"
              >
                About Me
              </a>
            </Link>,
            <Link
              href="/projects"
            >
              <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>
            </Link>,
            <Link
              href="/articles"
            >
              <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>
            </Link>,
            <Link
              href="/contact"
            >
              <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>
            </Link>,
          ],
          "className": "c-footer__nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                Home
              </a>,
              "href": "/index",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Home",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "Home",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                About Me
              </a>,
              "href": "/about",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "About Me",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "About Me",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>,
              "href": "/projects",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Projects",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "My Projects",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>,
              "href": "/articles",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Blog Articles",
                "className": "c-burgernav__menu",
              },
              "ref": null,
              "rendered": "Blog Articles",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>,
              "href": "/contact",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact",
                "className": "c-burgernav__menu u-white-border",
              },
              "ref": null,
              "rendered": "Contact",
              "type": "a",
            },
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="https://github.com/Dameon79"
            >
              <img
                className="c-socialmedialogos c-socialmedialogos__git"
                src="/static/images/Github2.png"
              />
            </a>,
            <a
              href="https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/procoders2.jpg"
              />
            </a>,
            <a
              href="https://www.linkedin.com/in/dameon-sumpton-407503ab/"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/linkedin1.png"
              />
            </a>,
            <a
              href="https://twitter.com/dameon_sumpton"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/Twitter1.png"
              />
            </a>,
          ],
          "className": "c-footer__socialmedia",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="c-socialmedialogos c-socialmedialogos__git"
                src="/static/images/Github2.png"
              />,
              "href": "https://github.com/Dameon79",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-socialmedialogos c-socialmedialogos__git",
                "src": "/static/images/Github2.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="c-socialmedialogos"
                src="/static/images/procoders2.jpg"
              />,
              "href": "https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-socialmedialogos",
                "src": "/static/images/procoders2.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="c-socialmedialogos"
                src="/static/images/linkedin1.png"
              />,
              "href": "https://www.linkedin.com/in/dameon-sumpton-407503ab/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-socialmedialogos",
                "src": "/static/images/linkedin1.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="c-socialmedialogos"
                src="/static/images/Twitter1.png"
              />,
              "href": "https://twitter.com/dameon_sumpton",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-socialmedialogos",
                "src": "/static/images/Twitter1.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Â© Dameon Sumpton 2018",
          "className": "c-footer__copyright",
        },
        "ref": null,
        "rendered": "Â© Dameon Sumpton 2018",
        "type": "h6",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="c-footer__nav"
          >
            <Link
              href="/index"
            >
              <a
                className="c-burgernav__menu"
              >
                Home
              </a>
            </Link>
            <Link
              href="/about"
            >
              <a
                className="c-burgernav__menu"
              >
                About Me
              </a>
            </Link>
            <Link
              href="/projects"
            >
              <a
                className="c-burgernav__menu"
              >
                My Projects
              </a>
            </Link>
            <Link
              href="/articles"
            >
              <a
                className="c-burgernav__menu"
              >
                Blog Articles
              </a>
            </Link>
            <Link
              href="/contact"
            >
              <a
                className="c-burgernav__menu u-white-border"
              >
                Contact
              </a>
            </Link>
          </nav>,
          <nav
            className="c-footer__socialmedia"
          >
            <a
              href="https://github.com/Dameon79"
            >
              <img
                className="c-socialmedialogos c-socialmedialogos__git"
                src="/static/images/Github2.png"
              />
            </a>
            <a
              href="https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/procoders2.jpg"
              />
            </a>
            <a
              href="https://www.linkedin.com/in/dameon-sumpton-407503ab/"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/linkedin1.png"
              />
            </a>
            <a
              href="https://twitter.com/dameon_sumpton"
            >
              <img
                className="c-socialmedialogos"
                src="/static/images/Twitter1.png"
              />
            </a>
          </nav>,
          <h6
            className="c-footer__copyright"
          >
            Â© Dameon Sumpton 2018
          </h6>,
        ],
        "className": "c-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href="/index"
              >
                <a
                  className="c-burgernav__menu"
                >
                  Home
                </a>
              </Link>,
              <Link
                href="/about"
              >
                <a
                  className="c-burgernav__menu"
                >
                  About Me
                </a>
              </Link>,
              <Link
                href="/projects"
              >
                <a
                  className="c-burgernav__menu"
                >
                  My Projects
                </a>
              </Link>,
              <Link
                href="/articles"
              >
                <a
                  className="c-burgernav__menu"
                >
                  Blog Articles
                </a>
              </Link>,
              <Link
                href="/contact"
              >
                <a
                  className="c-burgernav__menu u-white-border"
                >
                  Contact
                </a>
              </Link>,
            ],
            "className": "c-footer__nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  Home
                </a>,
                "href": "/index",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "Home",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  About Me
                </a>,
                "href": "/about",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "About Me",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "About Me",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  My Projects
                </a>,
                "href": "/projects",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Projects",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "My Projects",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu"
                >
                  Blog Articles
                </a>,
                "href": "/articles",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Blog Articles",
                  "className": "c-burgernav__menu",
                },
                "ref": null,
                "rendered": "Blog Articles",
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="c-burgernav__menu u-white-border"
                >
                  Contact
                </a>,
                "href": "/contact",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contact",
                  "className": "c-burgernav__menu u-white-border",
                },
                "ref": null,
                "rendered": "Contact",
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="https://github.com/Dameon79"
              >
                <img
                  className="c-socialmedialogos c-socialmedialogos__git"
                  src="/static/images/Github2.png"
                />
              </a>,
              <a
                href="https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE"
              >
                <img
                  className="c-socialmedialogos"
                  src="/static/images/procoders2.jpg"
                />
              </a>,
              <a
                href="https://www.linkedin.com/in/dameon-sumpton-407503ab/"
              >
                <img
                  className="c-socialmedialogos"
                  src="/static/images/linkedin1.png"
                />
              </a>,
              <a
                href="https://twitter.com/dameon_sumpton"
              >
                <img
                  className="c-socialmedialogos"
                  src="/static/images/Twitter1.png"
                />
              </a>,
            ],
            "className": "c-footer__socialmedia",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="c-socialmedialogos c-socialmedialogos__git"
                  src="/static/images/Github2.png"
                />,
                "href": "https://github.com/Dameon79",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-socialmedialogos c-socialmedialogos__git",
                  "src": "/static/images/Github2.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="c-socialmedialogos"
                  src="/static/images/procoders2.jpg"
                />,
                "href": "https://procoders.co.uk/?gclid=CjwKCAjwrNjcBRA3EiwAIIOvqwymZaA5y1EhNZgzLAr62QD6nkk3omRCfR4kU-2Xnpd9NrU9GGcknRoC-XkQAvD_BwE",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-socialmedialogos",
                  "src": "/static/images/procoders2.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="c-socialmedialogos"
                  src="/static/images/linkedin1.png"
                />,
                "href": "https://www.linkedin.com/in/dameon-sumpton-407503ab/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-socialmedialogos",
                  "src": "/static/images/linkedin1.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="c-socialmedialogos"
                  src="/static/images/Twitter1.png"
                />,
                "href": "https://twitter.com/dameon_sumpton",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-socialmedialogos",
                  "src": "/static/images/Twitter1.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Â© Dameon Sumpton 2018",
            "className": "c-footer__copyright",
          },
          "ref": null,
          "rendered": "Â© Dameon Sumpton 2018",
          "type": "h6",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
